dotnet/src/SemanticKernel.Abstractions/Orchestration/IKernelContext.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.SkillDefinition;
+
+namespace Microsoft.SemanticKernel.Orchestration;
+
+/// <summary>
+/// Kernel context reference
+/// </summary>
+public interface IKernelContext
+{
+    /// <summary>
+    /// App logger
+    /// </summary>
+    ILoggerFactory LoggerFactory { get; }
+
+    /// <summary>
+    /// Read only skills collection
+    /// </summary>
+    IReadOnlySkillCollection Skills { get; }
+
+    /// <summary>
+    /// Run a pipeline composed of synchronous and asynchronous functions.
+    /// </summary>
+    /// <param name="variables">Input to process</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <param name="pipeline">Target functions to run in sequence</param>
+    /// <returns>Result of the function composition</returns>
+    Task<SKContext> RunAsync(
+        ContextVariables variables,
+        CancellationToken cancellationToken = default,
+        params ISKFunction[] pipeline);
```

Comment:
Should we make the `cancellationToken` the last parameter in this new interface?

Reply:
The `params` has to be last... one of the reasons we're not a fan of this pipeline syntax. We'd like to replace this with a Plan-based approach and remove these overloads, but we have a few pre-reqs to get there. Roger and I discussed.

This is included for consistency with the (current) Kernel, but they'll all be replaced.

@RogerBarreto If you need this for something that exists in the codebase today, it's fine to keep it here to unblock. If you're adding this just for consistency, with Kernel and we have no callers in the repo, we should avoid adding it now (...and replacing it later).

Reply:
@markwallace-microsoft @shawncal 
I started this way TBH but I got lost trying to correct the IKernel to follow the same pattern and undid it.

Will give it another go.